### PR TITLE
[Docs] Remove deprecated methods and classes

### PR DESCRIPTION
**Pull request by jillnogold**
Ml-3232

### CODE REVIEW

File: docs/concepts/functions-overview.md [hunk 1]
```diff
@@ -38,10 +38,20 @@ common function methods:
...
-* with_limits(mem, cpu, gpus, gpu_type)
-* with_requests(mem, cpu)
 * set_env_from_secret(name, secret, secret_key)
 
+The limits methods are different for Spark and Dask:
+- Spark
+   - with_driver_limits(mem, cpu, gpu_type)
+   - with_executor_limits(mem, cpu, gpu_type)
+   - with_driver_resources(mem, cpu, gpu_type)
+   - with_executor_resources(mem, cpu, gpu_type)
```
Comment (alonmr) [inline docs/concepts/functions-overview.md:48]: `with_*_resources` is not a thing. remove please

File: docs/concepts/functions-overview.md [hunk 2]
```diff
@@ -38,10 +38,20 @@ common function methods:
...
+- Dask
+   - with_scheduler_limits(mem, cpu, gpu_type)
+   - with_worker_limits(mem, cpu, gpu_type)
+   - with_driver_resources(mem, cpu, gpu_type)
+   - with_worker_resources(mem, cpu, gpu_type)
```
Comment (alonmr) [inline docs/concepts/functions-overview.md:53]: same here

**Review by alonmr — CHANGES_REQUESTED** [review]
Looks great! few comments

File: docs/runtimes/dask-pipeline.ipynb [hunk 3]
```diff
@@ -39,8 +39,7 @@
      "name": "stdout",
```
Comment (alonmr) [inline docs/runtimes/dask-pipeline.ipynb:39]: <pre class="ql-syntax" spellcheck="false">Line #15.    def init_functions(functions: dict, project=None, secrets=None):
</pre><p>note to developers:</p><p><code>init_functions</code> are deprecated. we should fix the demo.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3056/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

File: docs/runtimes/mlrun_jobs.ipynb [hunk 4]
```diff
@@ -175,7 +175,7 @@
    "metadata": {},
```
Comment (alonmr) [inline docs/runtimes/mlrun_jobs.ipynb:175]: <pre class="ql-syntax" spellcheck="false">Line #1.   
mlrun.get_or_create_project(name='jobs-demo', "./")
_, artifact_path = mlrun.set_environment(artifact_path='./data/{{run.uid}}')
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3056/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

File: docs/concepts/functions-overview.md (hunk 1 above)
Comment (jillnogold) [reply docs/concepts/functions-overview.md:48]: removed

File: docs/concepts/functions-overview.md (hunk 2 above)
Comment (jillnogold) [reply docs/concepts/functions-overview.md:53]: removed

File: docs/runtimes/dask-pipeline.ipynb (hunk 3 above)
Comment (jillnogold) [reply docs/runtimes/dask-pipeline.ipynb:39]: there's a JIRAt: https://jira.iguazeng.com/browse/ML-3369

File: docs/runtimes/mlrun_jobs.ipynb (hunk 4 above)
Comment (jillnogold) [reply docs/runtimes/mlrun_jobs.ipynb:175]: updated

**Review by alonmr — APPROVED** [review]

**Pull request merged**